### DISCUSSION TITLE
Client-side LLM selection, multi-model support

### DISCUSSION THREAD

**Discussion by ThreeFish-AI** [langgenius/dify#4359] [Suggestion]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing?

Yes, our users hope that the client interface can support choosing the LLM they want to use.

### 2. Describe the feature you'd like to see

![4051715657913_ pic](https://github.com/langgenius/dify/assets/24806149/296a476f-2628-41c0-ba0e-174c6cc78d00)

It is expected to add the function of selecting LLM next to the input box similar to the chat bar.

### 3. How will this feature improve your workflow or experience?

We can experience the differences of multiple LLM versions in one Agent and choose the most appropriate LLM for use according to different scenarios.

### 5. Can you help us with this feature?

- [ ] I am interested in contributing to this feature.

**Comment by omesh845** [comment]
This is great, 😃 and It would be awesome if we had  the same feature via api also.

**Comment by perzeuss** [comment]
Basic Chatbots are designed to only have one LLM configured. In most cases you'll want it this way because every LLM behaves different and you would have to write and test different instructions.

If you're looking for a ChatGPT alternative with Dify assistants, this would be a different feature. This might come in the future but I think it is a different kind of product.

**Reply by ThreeFish-AI** [reply]
When using mainstream LLMs such as GPT-3.5, GPT-4, and Cloude-3, we may want to give users multiple choices by providing a single Agent. This is not just for testing, but sometimes for effects and reasons. Considering factors such as cost, you will need to choose the most appropriate model independently.
If each LLM configures an Agent, we will need to configure multiple identical Agents (only the LLM is different).
